cobalt/akn.py

```diff
@@ -129,11 +141,84 @@ class StructuredDocument(AkomaNtosoDocument):
     """ The xml of an empty document of the type self.document
     """
 
+    @classmethod
+    def for_document_type(cls, document_type):
+        """ Return the subclass for this document type.
+        """
+        def check_subclasses(klass):
+            for k in klass.__subclasses__():
+                if k.document_type and k.document_type.lower() == document_type:
+                    return k
+                # recurse
+                x = check_subclasses(k)
+                if x:
+                    return x
+
+        document_type = document_type.lower()
+        return check_subclasses(cls)
+
+    @classmethod
+    def empty_document(cls, version=DEFAULT_VERSION):
+        """ Return XML for an empty document of this type, using the given AKN version.
+        """
+        today = datestring(date.today())
+        frbr_uri = FrbrUri(
+            country='za',
+            locality=None,
+            doctype=cls.document_type,
+            subtype=None,
+            date=today,
+            number='1',
+            work_component='main',
+            language='eng',
+            actor=None,
+            prefix=('' if version == '2.0' else 'akn'),
```

Comment:
Will an empty string be exactly the same as `None` here? I'm worried that we'll end up with a URI that starts `//`.

Reply:
Yes, we do `if self.prefix:`